src/exchanges/crdtExchange.ts

```diff
@@ -258,42 +261,29 @@ class CrdtManager {
   #client: Client;
   #next: (op: Operation) => void;
   #options: CrdtExchangeOpts;
+  #storage: DefaultStorage;
 
   #mutations = new Map<MutationOperation['key'], MutationOperation>();
   #queries = new Map<QueryOperation['key'], QueryOperation>();
 
   #optimisticState: MergedCrdtObjectStore = new Map();
   #unsubscribe: null | (() => void) = null;
 
+  hydration: Promise<void>;
+
   constructor(client: Client, next: (op: Operation) => void, options: CrdtExchangeOpts) {
     this.#client = client;
     this.#next = next;
     this.#options = options;
+    this.#storage = options.storage;
+
+    this.hydration = this.rehydrate();
   }
 
-  // TODO allow adding mutations in bulk so we can update optimistic state and execute dependent queries only once
   addMutation(mutation: MutationOperation) {
     this.#mutations.set(mutation.key, mutation);
-
-    this.updateOptimisticState([mutation]);
-
-    const dependentQueries = new Set<QueryOperation>();
-    this.collectDependentQueries(dependentQueries, mutation);
-    for (const dependentQuery of dependentQueries) {
-      this.#next(toRequestPolicy(dependentQuery, 'cache-only'));
-    }
-
-    if (!this.#unsubscribe) {
-      this.#unsubscribe = pipe(
-        interval(10000),
-        subscribe((n) => {
-          console.log(`processing mutations ${n}...`);
-          // TODO: we should send at a more rapid pace when stuck doing only one mutation a time
-          this.sendMutations();
-        }),
-      ).unsubscribe;
-    }
-    return this;
+    this.persistMutations();
```

Comment:
Why do we persist before processing? Would it be better to persist afterward so we can capture the new mutation?

Reply:
We are persisting the contents of the `#mutations` field, in order to ensure that state can be restored if app is killed. Therefore, the time to persist is right after `#mutations` is mutated.

`processMutations` doesn't mutate `#mutations` so it makes sense to just persist the new mutation as soon as it's added to `#mutations`.

It appears that I missed one of the places in this class where `#mutations` is mutated, which is in `applyMutation`. We need to be persisting `#mutations` there as well.

Reply:
Ah, I was confused with the diff - I see now that we `this.#mutations.set` _then_ `persistMutations`. Looks fine.